### PR TITLE
HW - Hb 6

### PR DESCRIPTION
**Pull request by ilya449**

### CODE REVIEW

File: src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> query = session.createQuery("FROM ShoppingCart "
+                    + "WHERE id = :userId ", ShoppingCart.class);
```
Comment (vremere-a) [inline src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java:42]: join fetch?

File: src/main/java/com/cinema/model/ShoppingCart.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import lombok.RequiredArgsConstructor;
+
+@Data
+@RequiredArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (vremere-a) [inline src/main/java/com/cinema/model/ShoppingCart.java]: if you use MapID 
you don't need @GeneratedValue(strategy = GenerationType.IDENTITY)

File: src/main/java/com/cinema/service/ShoppingCartService.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import com.cinema.model.MovieSession;
+import com.cinema.model.ShoppingCart;
+import com.cinema.model.User;
+
+public interface ShoppingCartService {
+    /**
+     * This method is responsible to add a Ticket to the ShoppingCart
+     * @param movieSession Contains the information required for a ticket
+     * @param user - user who wan't to buy a ticket for a specific MovieSession
+     */
```
Comment (vremere-a) [inline src/main/java/com/cinema/service/ShoppingCartService.java]: you need del this

File: src/main/java/com/cinema/model/ShoppingCart.java (hunk 2 above)
Comment (ilya449) [reply src/main/java/com/cinema/model/ShoppingCart.java]: Ok, thanks

File: src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java (hunk 1 above)
Comment (ilya449) [reply src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java:42]: I don't need it, I have 
    @OneToMany(fetch = FetchType.EAGER)
    private List<Ticket> tickets;  
and it works well

Comment (vremere-a) [reply src/main/java/com/cinema/dao/impl/ShoppingCartDaoImpl.java:42]: 👍

File: src/main/java/com/cinema/service/impl/ShoppingCartServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import com.cinema.model.ShoppingCart;
+import com.cinema.model.Ticket;
+import com.cinema.model.User;
+import com.cinema.service.ShoppingCartService;
+import java.util.ArrayList;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    ShoppingCartDao shoppingCartDao;
```
Comment (Sofasmile) [inline src/main/java/com/cinema/service/impl/ShoppingCartServiceImpl.java]: ?

File: src/main/java/com/cinema/model/ShoppingCart.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+@Data
+@RequiredArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    private Long id;
+    @NonNull
+    @OneToMany(fetch = FetchType.EAGER)
```
Comment (Sofasmile) [inline src/main/java/com/cinema/model/ShoppingCart.java]:
```suggestion
    @OneToMany
```
не використовуємо FetchType.EAGER

File: src/main/java/com/cinema/service/impl/ShoppingCartServiceImpl.java (hunk 4 above)
Comment (ilya449) [reply src/main/java/com/cinema/service/impl/ShoppingCartServiceImpl.java]: added 'private'

File: src/main/java/com/cinema/model/ShoppingCart.java (hunk 5 above)
Comment (ilya449) [reply src/main/java/com/cinema/model/ShoppingCart.java]: ok, done with out EAGER

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**